src/main/java/com/wanderersoftherift/wotr/core/rift/RiftEvents.java

```diff
@@ -22,9 +23,11 @@ public static void onRiftTick(LevelTickEvent.Post event) {
             for (var playerID : data.getPlayers()) {
                 var player = serverLevel.getServer().getPlayerList().getPlayer(playerID);
                 if (player == null) {
-                    continue; // Player is offline
+                    // Player is offline
+                    continue;
                 }
                 if (player.isDeadOrDying()) {
+                    NeoForge.EVENT_BUS.post(new RiftEvent.PlayerDied(player, serverLevel, data.getConfig()));
                     toRemove.add(playerID);
                 }
                 if (player.level() instanceof ServerLevel pLevel && !RiftData.isRift(pLevel) && pLevel != serverLevel) {
```

Comment:
The RiftData check here is obsolete, as we are already returning at the top if it is not a rift.

Reply:
Gone with the refactor